framework/app.mk

```diff
@@ -159,13 +159,14 @@ endif
 
 BUILD_EXEC :=
 
-$(app_objects): $(app_HEADER)
-
+app_GIT_DIR := $(shell cd "$(APPLICATION_DIR)" && git rev-parse --show-toplevel)
```

Comment:
This command would fail if you downloaded a tarball of MOOSE which is a possibility. Maybe if it's blank and we depend on blank nothing happens? Maybe put in a garbage command and see if it does something bad? I wonder how many other things totally fail if we don't have a git repo. I guess you probably can't build libmesh either because of our script? Hmm... Maybe it doesn't mater for now